src/modules/announcements/pages/Announcements.jsx

```diff
@@ -1,3 +1,53 @@
+import { useEffect, useState } from "react";
+import { useLocation } from "react-router-dom";
+import { SearchForm } from "../../../components/SearchForm/SearchForm";
+import { ANNOUNCEMENTS_LIST } from "../../../constants/announcements";
+
 export const AnnouncementList = () => {
-  return <div>announcement list</div>;
+  const [filteredAnnouncements, setFilteredAnnouncements] = useState([]);
+  const { search } = useLocation();
+
+  const filterAnnouncements = (data) => {
+    const { Phrase, Category, City, Animals } = data;
+    let announcements = [...ANNOUNCEMENTS_LIST];
+    let animalsArray;
+
+    announcements = announcements
+      .filter((announcement) =>
+        announcement.title.toLowerCase().includes(Phrase.toLowerCase()),
+      )
+      .filter((announcement) =>
+        announcement.category.toLowerCase().includes(Category),
+      )
+      .filter((announcement) =>
+        announcement.city.toLowerCase().includes(City.toLowerCase()),
+      );
+    if (Animals) {
+      animalsArray = Animals.map((animal) => animal.toLowerCase());
+      announcements = announcements.filter((announcement) =>
+        animalsArray.includes(announcement.animal.toLowerCase()),
+      );
+    }
+
+    setFilteredAnnouncements(announcements);
```

Comment:
If you do that all playground with `announcements` just to set it in `state` then it should be in a `useEffect(() => { ... }, [announcements])` or even in a function outside the component.